Tests/SourceControlTests/GitRepositoryTests.swift

```diff
@@ -736,4 +737,86 @@ class GitRepositoryTests: XCTestCase {
             XCTAssertNoThrow(try checkoutRepo.getCurrentRevision())
         }
     }
+
+    func testErrorMessagesIncludeExitStatusIfNoOtherDataIsAvailable() throws {
+        try testWithTemporaryDirectory { path in
+            let testRepoPath = path.appending(component: "test-repo")
+            try makeDirectories(testRepoPath)
+            initGitRepo(testRepoPath)
+
+            let git = ShellHelperStub()
+            let repo = GitRepository(git: git, path: testRepoPath)
+
+            git.exitStatus = .terminated(code: 128)
+
+            XCTAssertThrowsError(try repo.fetch(), "") { error in
+                XCTAssertEqual(String(describing: error), """
+                Couldn’t fetch updates from remote repositories:
+                    Git process terminated with status 128
+                """)
+            }
+
+            #if os(Windows)
+            git.exitStatus = .abnormal(exception: 2)
```

Comment:
I'm not able to test this, as I don't have a Windows setup. Would it run on CI?

Reply:
Unfortunately, no, CI only builds for Windows but currently does not run the test suite.